prometheus/promhttp/option.go

```diff
@@ -24,14 +26,32 @@ type Option interface {
 	apply(*options)
 }
 
+// LabelResolver are used to compute the label value from request or response.
+// Response will be provided only when used on a round tripper
+type LabelResolver func(*http.Request, *http.Response) string
```

Comment:
Hm, to be consistent. Ideally, we want to have RespWriter here as well - which might mean separate abstractions. Also, this is not ideal as somebody might want to edit those as well (I guess we don't have any option here - we can't mark them for read-only - unless we want to somehow pack them in interfaces, which might be a fuss).

That's why in the way... https://github.com/prometheus/client_golang/pull/1104 change reduces the surface of error - one has to put some value in to context in extra middleware if they want it to get some label in and makes this dynamic function read only. It also means more code to user.